File: sunoscribe/backend/app/schemas/task.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class TaskStatusDTO(BaseModel):
    task_id: uuid.UUID
    project_id: uuid.UUID
    task_type: str
    transcription_target: str = "lead_vocal"
    status: str
    progress: int = Field(ge=0, le=100)
    retry_count: int = Field(ge=0)
    max_retries: int = Field(ge=0)
    can_retry: bool
    failure_reason: str | None = None
    error_message: str | None = None
    result_payload: dict[str, Any] = Field(default_factory=dict)
    queued_at: datetime
    started_at: datetime | None = None
    finished_at: datetime | None = None


class TaskCreateDTO(BaseModel):
    task_id: uuid.UUID
    project_id: uuid.UUID
    task_type: str
    transcription_target: str = "lead_vocal"
    status: str
    progress: int = Field(ge=0, le=100)
    retry_count: int = Field(ge=0)
    max_retries: int = Field(ge=0)
    can_retry: bool = False
    failure_reason: str | None = None
    error_message: str | None = None
    queued_at: datetime
    started_at: datetime | None = None
    finished_at: datetime | None = None

File: sunoscribe/frontend/src/types/task.ts
export type TaskStatus = 'queued' | 'running' | 'succeeded' | 'failed' | 'retrying'

export interface TaskStatusResponse {
  task_id: string
  project_id: string
  task_type: string
  status: TaskStatus | string
  progress: number
  retry_count: number
  max_retries: number
  can_retry: boolean
  error_message: string | null
  result_payload: Record<string, unknown>
  queued_at: string | null
  started_at: string | null
  finished_at: string | null
}

